my_enumerable.rb

```diff
@@ -0,0 +1,198 @@
+# rubocop:disable Style/CaseEquality, Metrics/ModuleLength, Style/For, Lint/RedundantCopDisableDirective
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for element in self
+      yield(element)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    return to_enum(:my_each_with_index) unless block_given?
```

Comment:
[Optional]

I think it would be better to return the `to_enum` before the `i = 0` assignment since it will be useless when no block is given.

Reply:
done